rwgame/RWConfig.cpp

```diff
@@ -89,39 +81,62 @@ constexpr std::string_view arg_mask_to_key(std::string_view v) {
     return v.substr(maxstart, maxsize);
 }
 
+template <typename T>
+std::optional<T> vm_get_opt_argmask(const po::variables_map& vm, const std::string_view& argmask) {
+    const std::string key{arg_mask_to_key(argmask)};
+    auto it = vm.find(key);
+    if (it != vm.end()) {
+        return it->second.as<T>();
+    }
+    return std::nullopt;
+}
+
+template <>
+std::optional<bool> vm_get_opt_argmask<bool>(const po::variables_map& vm, const std::string_view& argmask) {
+    const std::string key{arg_mask_to_key(argmask)};
+    auto it = vm.find(key);
+    if (it != vm.end()) {
+        return true;
+    }
+    return std::nullopt;
```

Comment:
nit(?): `false`?
`&& *val;` of line 106 looks quite unneeded.

Reply:
No. If it is not present, it can be overridden by a config or default value.